selfdrive/updated.py

```diff
@@ -203,13 +209,49 @@ def attempt_update():
     if new_version:
       cloudlog.info("git reset in progress")
       r = [
-        run(NICE_LOW_PRIORITY + ["git", "reset", "--hard", "@{u}"], OVERLAY_MERGED),
-        run(NICE_LOW_PRIORITY + ["git", "clean", "-xdf"], OVERLAY_MERGED),
-        run(NICE_LOW_PRIORITY + ["git", "submodule", "init"], OVERLAY_MERGED),
-        run(NICE_LOW_PRIORITY + ["git", "submodule", "update"], OVERLAY_MERGED),
+        run(["git", "reset", "--hard", "@{u}"], OVERLAY_MERGED, low_priority=True),
+        run(["git", "clean", "-xdf"], OVERLAY_MERGED, low_priority=True ),
+        run(["git", "submodule", "init"], OVERLAY_MERGED, low_priority=True),
+        run(["git", "submodule", "update"], OVERLAY_MERGED, low_priority=True),
       ]
       cloudlog.info("git reset success: %s", '\n'.join(r))
 
+      # Download the accompanying NEOS version if it doesn't match the current version
+      with open(NEOS_VERSION, "r") as f:
+        current_neos_version = f.read().strip()
+
+      required_neos_version = run(["bash", "-c",
+                                   r"unset REQUIRED_NEOS_VERSION && source launch_env.sh && echo -n $REQUIRED_NEOS_VERSION"],
+                                   OVERLAY_MERGED).strip()
+
+      cloudlog.info(f"NEOS version update check: {current_neos_version} current, {required_neos_version} in update")
+      if current_neos_version != required_neos_version:
+        cloudlog.info(f"Beginning background download for NEOS {required_neos_version}")
+
+        update_manifest = f'file:///{OVERLAY_MERGED}/installer/updater/update.json'
+        set_offroad_alert("Offroad_NeosUpdate", True)
+
+        neos_downloaded = False
+        start_time = time.monotonic()
+        # Try to download for one day
+        while (time.monotonic() - start_time < 60*60*24) and not wait_helper.shutdown:
+          try:
+            updater_path = os.path.join(OVERLAY_MERGED, "installer/updater/updater")
+            run([updater_path, "bgcache", update_manifest], OVERLAY_MERGED, low_priority=True)
```

Comment:
Is there a way to kill the process if `wait_helper.shutdown` triggers? This can take a very long otherwise.

Reply:
fixed this for the git fetch/pull and the NEOS update case. when wait_helper.shutdown gets set, it'll forward 
SIGTERM/SIGINT to any child processes